src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala

```diff
@@ -25,92 +25,135 @@ import upickle.default.write
 import scala.reflect.ClassTag
 import scalaz.concurrent.Task
 
-case class GetManyDocumentsQueryBuilder(
+sealed trait DocsInResult
+abstract class IncludeDoc[D: R] extends DocsInResult
+abstract class ExcludeDocs extends DocsInResult
+
+sealed trait MissingIdsInQuery
+trait MissingAllowed extends MissingIdsInQuery
+trait MissingNotAllowed extends MissingIdsInQuery
+
+sealed trait DocType
+abstract class ForDocType[D: R, K: R, V: R] extends DocType
+abstract class AnyDocType extends DocType
+
+case class GetManyDocumentsQueryBuilder[ID <: DocsInResult, AM <: MissingIdsInQuery, BT <: DocType](
     client: Client,
     db: String,
     typeMappings: TypeMapping,
-    params: Map[String, String] = Map.empty[String, String]) extends QueryStrategy {
+    params: Map[String, String] = Map.empty[String, String],
+    ids: Seq[String] = Seq.empty, view: CouchView = QueryUtils.tempTypeFilterView) {
 
   private val log = org.log4s.getLogger
 
-  lazy val tempTypeFilterView: CouchView = {
-    CouchView(
-      map =
-          """
-            |function(doc) {
-            | emit([doc.kind, doc._id], doc._id);
-            |}
-          """.stripMargin)
+  def conflicts(
+      conflicts: Boolean = true): GetManyDocumentsQueryBuilder[ID, AM, BT] = {
+    set[ID, AM, BT]("conflicts", conflicts)
+  }
+
+  def descending(
+      descending: Boolean = true): GetManyDocumentsQueryBuilder[ID, AM, BT] = {
+    set[ID, AM, BT]("descending", descending)
+  }
+
+  def endKey[K: W](endKey: K): GetManyDocumentsQueryBuilder[ID, AM, BT] = {
+    set[ID, AM, BT]("endkey", write(endKey))
+  }
+
+  def endKeyDocId(
+      endKeyDocId: String): GetManyDocumentsQueryBuilder[ID, AM, BT] = {
+    set[ID, AM, BT]("endkey_docid", endKeyDocId)
   }
 
-  def conflicts(conflicts: Boolean = true): GetManyDocumentsQueryBuilder = {
-    set("conflicts", conflicts)
+  def includeDocs[D: R]: GetManyDocumentsQueryBuilder[IncludeDoc[D], AM, BT] = {
```

Comment:
I initially thought it might be a bit redundant since these are always turned off by default `GetManyDocumentsQueryBuilder[ExcludeDocs, MissingNotAllowed, AnyDocType]` and thus require explicit calls to turn these on, which makes scenario that you would want turn it backoff a bit less likely? ... but could still be useful I guess ... adding overloaded methods for this so as to still have the convenience of not having to specify braces :-)